### ISSUE TITLE
Add ignore: ["selectors-within-list"] to no-descending-specificity

### ISSUE THREAD

**Issue by sindresorhus** [stylelint/stylelint#3196]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

> Which rule, if any, is this issue related to?

`no-descending-specificity`

> What CSS is needed to reproduce this issue?

```css
a {
	color: #000;
}

a:hover {
	text-decoration: underline;
}

h1,
h2,
h3,
h4,
h5,
h6,
a,
button {
	user-select: none;
	cursor: default;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

> Which version of stylelint are you using?

9.1.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

I expected it to not trigger the `no-descending-specificity` rule as it's common to write CSS like this.

> What actually happened (e.g. what warnings or errors you are getting)?

It triggered the rule on the `a,`. It would be unnatural to move the `a:hover` below the `...h5, h6, a,...`.

I don't think it should trigger when the selector is part of multiple selectors. Or there should at least be an option to achieve this.

**Comment by jeddy3** [comment]
> I don't think it should trigger when the selector is part of multiple selectors. Or there should at least be an option to achieve this.

This sounds good to me. Adding an option to make a rule more permissive (especially when the pattern is common) is our common approach.

I don't think we've added an `ignore` option for anything similar in other rules, so I'm not 100% sure how the option should be named. We have quite a few `ignore: ["inside-*"]` options e.g. `"inside-block"`, should we use "inside" here or is there a clearer name e.g.

- `ignore: ["inside-selector-list"]`
- `ignore: ["selectors-within-list"]`

I'm currently favouring the latter as the existing "inside" options belong to rules that are scoped to _things_ within the source rather than the source itself. e.g. `rule-empty-line-before: ["always", { ignore: ["inside-block"] }]` is scoped to rules. Whereas, this rule is called `no-descending-specificity` as it is scoped to the source, as such I think we need to identify the thing being ignored with the name of the option e.g. selectors within (selector) lists.

**Comment by jeddy3** [comment]
As there don't seem to be any objections to the proposal of a new option, I'm going to label this up accordingly.
